code/modules/atmospherics/machinery/components/fusion/hfr_procs.dm

```diff
@@ -446,6 +513,29 @@
 
 	qdel(src)
 
+/**
+ * Induce hallucinations in nearby humans.
+ *
+ * force will make hallucinations ignore meson protection.
+ */
+/obj/machinery/atmospherics/components/unary/hypertorus/core/proc/induce_hallucination(strength, delta_time, force=FALSE)
+	for(var/mob/living/carbon/human/human in view(src, HALLUCINATION_HFR(heat_output)))
+		if(force || !istype(human.glasses, /obj/item/clothing/glasses/meson))
+			var/distance_root = sqrt(1 / max(1, get_dist(human, src)))
+			human.hallucination += strength * distance_root * delta_time
+			human.hallucination = clamp(human.hallucination, 0, 200)
```

Comment:
This should be an early continue. Don't like that it's close to the same behavior as the sm, but that's for another time

Reply:
Gotcha

Reply:
Done